src/main/java/edu/kit/datamanager/security/filter/NoAuthenticationFilter.java

```diff
@@ -73,7 +73,7 @@ protected void doFilterInternal(HttpServletRequest request, HttpServletResponse
       claimMap.put(entry.getKey(), entry.getValue());
     });
 
-    String token = Jwts.builder().setClaims(claims).setExpiration(DateUtils.addHours(new Date(), 1)).signWith(SignatureAlgorithm.HS512, secretKey).compact();
+    String token = Jwts.builder().setClaims(claims).setExpiration(DateUtils.addHours(new Date(), 1)).signWith(SignatureAlgorithm.HS256, secretKey).compact();
```

Comment:
Is this signature algorithm mandatory or should we make this configurable?
Sounds like an 'old' algorithm.

Reply:
I think it's good enough and it is only for self-generated keys, which should be only used rarely. Typically, keycloak will be used and there this line does not apply in any case.

Reply:
Sure, but why did you change this if it is not that important?
Old algorithms may be deprecated/insecure sooner or later.

Reply:
Because I only had a 64 character random string in my configuration file and HS512 requires an even longer string. As far as I remember, keycloak is also using HS256 so I stayed with this.